### ISSUE TITLE
SC2312 check-extra-masked-returns should be warning and default

### ISSUE THREAD

**Issue by bruce-ricard** [koalaman/shellcheck#2807]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

[SC2312](https://github.com/koalaman/shellcheck/wiki/SC2312) is currently an `info` level check, and an optional one. 
The [documentation page](https://www.shellcheck.net/wiki/Optional) about optional checks states that
> These are generally subjective or stylistic.

It seems to me that avoiding masking the return value of a subcommand is extremely important. Not only it should not be optional, but it should be of `warning` severity. It looks like it should be treated the same was as [SC2155](https://github.com/koalaman/shellcheck/wiki/SC2155) "Declare and assign separately to avoid masking return values.", which is a default warning.
They both try to catch the same general issue: the masking of return values.

I recently realized that one of my scripts had been silently failing for months because a return value was masked in a subcommand.

Please let me know if you see things differently.

cc: @DoxasticFox who worked on the check

**Comment by kurahaupo** [comment]
I agree that this should be treated more severely, but I wouldn't change the rating of [SC2155](../wiki/SC2155), since `local` or `declare` can't by itself cause havoc.

Both the common cases involving `cd` are already caught:

1. Shellcheck already gives warning [SC2046](../wiki/SC2046) about unquoted `$(cmd)`
2. For `cd "$(cmd)"`, if `cmd` fails without any output, the resulting `cd ''` fails (with status 1).

So it seems like you were unlucky to have one of the uncommon cases:

3. Either your `cmd` somehow outputs a valid path, despite failing (perhaps it outputs a sequence of lines like `/dir1`, `/dir2` etc, but dies before completing the sequence); or
4. You have `cd "/valid/path/$(cmd)"` and `cmd` fails with _no_ output; or
5. Perhaps you have `# shellcheck ignore=SC2046` before `cd ...`. Ahem.

To cover cases (1) and (2), I would add high-severity warnings specifically for `cd` with expansions:

* for `cd $(cmd)`, all the caveats about unquoted variable and parameter expansions apply, but in addition, if `cmd` produces no output, `cd $(cmd)` will be equivalent to `cd "$HOME"`.
* for `cd "$( for p in /dir1 /dir2 /dir3 ... ; do echo $p ; do_something_bad_and_die ; done )"`, this needs a clone of [SC2807](../wiki/SC2807] that's at "warning" or "error" level.
* for `cd /valid/path/"$(cmd)"`, if `cmd` fails and therefore produces no output, `cd /valid/path/` will _succeed_. That would also be caught by the same new error.

There's no fix for case (53) besides saying "don't do that".
